File: HomeWork 2/flower.cpp
//------------------------------------------------------------------------------
// flower.cpp - содержит функции обработки цветка
//------------------------------------------------------------------------------

#include "flower.h"

//------------------------------------------------------------------------------
// Ввод параметров цветка из потока
void flower::In(std::ifstream &ifst) {
    int place;
    ifst >> place >> this->name;
    switch (place) {
        case 1:
            this->flow_type = flower::flower_type::DOMESTIC;
            break;
        case 2:
            this->flow_type = flower::flower_type::GARDEN;
            break;
        case 3:
            this->flow_type = flower::flower_type::WILD;
            break;
        default:
            this->flow_type = flower::flower_type::INVALID_PLACE_NUMBER;
    }
}

// Случайный ввод параметров цветка
void flower::InRnd() {
    switch (Random() % 3) {
        case 0:
            this->flow_type = flower::flower_type::DOMESTIC;
            break;
        case 1:
            this->flow_type = flower::flower_type::GARDEN;
            break;
        case 2:
            this->flow_type = flower::flower_type::WILD;
            break;
        default:
            this->flow_type = flower::flower_type::INVALID_PLACE_NUMBER;
    }
    for (size_t i = 0; i < MAX_NAME_SIZE; ++i) {
        this->name[i] = static_cast<char>('a' + Random());
    }
    this->name[MAX_NAME_SIZE - 1] = '\0';
}

//------------------------------------------------------------------------------
// Вывод параметров цветка в поток
void flower::Out(std::ofstream &ofst) {
    ofst << "It is flower: flower type = ";
    switch (this->flow_type) {
        case flower::DOMESTIC:
            ofst << "domestic";
            break;
        case flower::GARDEN:
            ofst << "garden";
            break;
        case flower::WILD:
            ofst << "wild";
            break;
        case flower::INVALID_PLACE_NUMBER:
            ofst << "Invalid place number!";
    }
    ofst << ", name = " << this->name
         << ". CountExpr = " << this->CountExpr() << "\n";
}

//------------------------------------------------------------------------------
// Вычисление частного от деления числа гласных букв
// в названии цветка на общую длину его названия

double flower::CountExpr() {
    // переменная для подсчёта гласных в названии цветка
    size_t number_of_vowels = 0;
    for (auto it: this->name) {
        char current_symbol = tolower(it);
        if (current_symbol == 'a' || current_symbol == 'e' || current_symbol == 'i' ||
            current_symbol == 'o' || current_symbol == 'u' || current_symbol == 'y') {
            ++number_of_vowels;
        }
    }
    return (double)number_of_vowels / strlen(this->name);
}
flower::~flower() {
    
}

File: HomeWork 3/flower.py
from enum import Enum
from rnd import Random, randName

class FlowerType(Enum):
    DOMESTIC = 1
    GARDEN = 2
    WILD = 3
    INVALID_PLACE_NUMBER = 0

# Класс цветка
class Flower :
    # Конструктор цветка
    def __init__(self):
        self.name = ""
        self.flowerType = FlowerType.INVALID_PLACE_NUMBER.name
    
    # Считывание параметров цветка из строки
    def ReadStrArray(self, strArray, i):
        if i >= len(strArray) - 1:
            return 0
        self.name = strArray[i]
        self.flowerType = FlowerType(int(strArray[i + 1]) 
            if int(strArray[i + 1]) <= 3 and int(strArray[i + 1]) >= 0 else 0).name
        i += 2
        return i

    # Случайная генерация параметров цветка
    def RandomIn(self):
        self.name = randName()
        self.flowerType = FlowerType(Random() % 3 + 1).name

    # Вывод данных цветка в файл
    def Write(self, ostream):
        ostream.write(f"Flower: name = {self.name}, flower type = {self.flowerType}, CountExpr = {self.CountExpr()}")

    # Вычисление частного от деления числа гласных букв
    # в названии цветка на общую длину его названия
    def CountExpr(self):
        numberOfVowels = 0
        for ch in self.name:
            if(ch == 'a' or ch == 'e' or ch == 'i' or ch == 'o' or ch == 'u' or ch == 'A'
                or ch == 'E' or ch == 'I' or ch == 'O' or ch == 'U'):
                numberOfVowels += 1
        return numberOfVowels / len(self.name)
    